aiidalab_qe/node_view.py

```diff
@@ -108,74 +108,84 @@ def _update_structure_viewer(self, change):
-                kind_name = orbital_data["kind_name"]
-                orbital_name = orbital.get_name_from_quantum_numbers(
-                    orbital_data["angular_momentum"], orbital_data["magnetic_number"]
-                )
-                if suffix is not None:
-                    orbital_name += f"-{suffix}"
-
-                pdos_orbitals.append(
-                    {
-                        "kind": kind_name,
-                        "orbital": orbital_name,
-                        "energy | eV": energy,
-                        "pdos | states/eV": pdos,
-                    }
-                )
+    if "projections" in work_chain_node.outputs:
+        projection_list = [
+            (work_chain_node.outputs.projections, None),
+        ]
+    else:
+        projection_list = [
+            (work_chain_node.outputs.projections_up, "up"),
+            (work_chain_node.outputs.projections_down, "dn"),
+        ]
+        tdos_values["dos | states/eV"] = tdos_values.pop(
+            "dos_spin_up | states/eV"
+        ) + tdos_values.pop("dos_spin_down | states/eV")
+
+    for projections, suffix in projection_list:  # type: ProjectionData, str
+        for orbital, pdos, energy in projections.get_pdos():
+            orbital_data = orbital.get_orbital_dict()
+            kind_name = orbital_data["kind_name"]
+            orbital_name = orbital.get_name_from_quantum_numbers(
+                orbital_data["angular_momentum"], orbital_data["magnetic_number"]
+            )
+            if suffix is not None:
+                orbital_name += f"-{suffix}"
+
+            pdos_orbitals.append(
+                {
+                    "kind": kind_name,
+                    "orbital": orbital_name,
+                    "energy | eV": energy,
+                    "pdos | states/eV": pdos,
+                }
+            )
 
-        data_dict = {
-            "fermi_energy": fermi_energy,
-            "tdos": {f"energy | {energy_units}": energy_dos, "values": tdos_values},
-            "pdos": pdos_orbitals,
-        }
+    data_dict = {
+        "fermi_energy": fermi_energy,
+        "tdos": {f"energy | {energy_units}": energy_dos, "values": tdos_values},
+        "pdos": pdos_orbitals,
+    }
 
-        # And this is why we shouldn't use special encoders...
-        return json.loads(json.dumps(data_dict, cls=MontyEncoder))
+    # And this is why we shouldn't use special encoders...
+    return json.loads(json.dumps(data_dict, cls=MontyEncoder))
 
 
 def export_data(work_chain_node):
+    if "dos" in work_chain_node.outputs:
+        fermi_energy = work_chain_node.outputs.nscf_parameters["fermi_energy"]
+        dos = export_pdos_data(work_chain_node, fermi_energy)
+    else:
+        fermi_energy = None
+        dos = None
+
+    if "band_structure" in work_chain_node.outputs:
+        # using the fermi energy from dos nscf if calculated.
+        bands = export_bands_data(work_chain_node, fermi_energy)
+    else:
+        bands = []
```

Comment:
If there is no band data, set it to `None`.
```suggestion
        bands = None
```

Reply:
Yes, that's true. Thanks!